src/components/views/dialogs/HostSignupDialog.tsx

```diff
@@ -0,0 +1,266 @@
+    }
+
+    private onCloseClick = async () => {
+        if (this.state.completed) {
+            // We're done, close
+            return this.closeDialog();
+        } else {
+            Modal.createDialog(
+                QuestionDialog,
+                {
+                    title: _t("Confirm Abort Of Host Creation"),
+                    description: _t(
+                        "Are you sure you wish to abort creation of the host? The process cannot be continued.",
+                    ),
+                    button: _t("Abort"),
+                    onFinished: result => {
+                        if (result) {
+                            return this.closeDialog();
+                        }
+                    },
+                },
+            );
+        }
+    }
+
+    private sendMessage = (message: IPostmessageResponseData) => {
+        this.iframeRef.current.contentWindow.postMessage(message, this.config.url);
+    }
+
+    private async sendAccountDetails() {
+        const openIdToken = await MatrixClientPeg.get().getOpenIdToken();
+        if (!openIdToken || !openIdToken.access_token) {
+            this.setState({
+                completed: true,
+                error: _t("Failed to connect to your homeserver. Please close this dialog and try again."),
+            });
+            return;
+        }
+        this.sendMessage({
+            action: PostmessageAction.HostSignupAccountDetails,
+            account: {
+                accessToken: await MatrixClientPeg.get().getAccessToken(),
+                name: OwnProfileStore.instance.displayName,
+                openIdToken: openIdToken.access_token,
+                serverName: await MatrixClientPeg.get().getDomain(),
+                userLocalpart: await MatrixClientPeg.get().getUserIdLocalpart(),
+                termsAccepted: true,
+            },
+        });
+    }
+
+    private onAccountDetailsDialogFinished = async (result) => {
+        if (result) {
+            return this.sendAccountDetails();
+        }
+        return this.closeDialog();
+    }
+
+    private onAccountDetailsRequest = () => {
+        const termsDialog = this.config.termsDialog;
+        const textComponent = (
+            <>
+                <p>
+                    {termsDialog.text}
+                </p>
+                <p>
+                    {_t("Learn more in our")}&nbsp;
+                    <a href={termsDialog.privacyPolicy.href} target="_blank" rel="noreferrer noopener">
+                        {termsDialog.privacyPolicy.text}
+                    </a>,&nbsp;
+                    <a href={termsDialog.termsOfService.href} target="_blank" rel="noreferrer noopener">
+                        {termsDialog.termsOfService.text}
+                    </a> {_t("and")}&nbsp;
+                    <a href={termsDialog.cookiePolicy.href} target="_blank" rel="noreferrer noopener">
+                        {termsDialog.cookiePolicy.text}
+                    </a>.
```

Comment:
This needs re-building with i18n in mind. Our translations can take string/component interpolations.
Translating individual words e.g `and` doesn't work in many languages to keep the sense of the sentence.

Reply:
Have rebuilt the whole thing to properly use i18n.